### ISSUE TITLE
how exactly do you install this?

### ISSUE THREAD

**Issue by gstlouisgit** [Gcenx/WineskinServer#33]
I fell on this because I can't get wineskin working on my macbook running high sierra.  But this build looks different and I do not see any install instructions

Can someone explain how to install this?

thanks

**Comment by AkkuDakku** [comment]
https://www.youtube.com/watch?v=2f9vROQJ-9c

**Comment by Gcenx** [comment]
@AkkuDakku you shouldn’t run an application from download that causes translocation issues.

The Winery application should be installed to ~/Applications not /Applications as Winery will self modify on update.

The explanation of setting the correct boot-arg(10.15.4+)/SIP disabled to function is missing.

For Engine you don’t need to use the entire list instead use the search box above and enter WS11 to show the engines.

**Comment by gstlouisgit** [comment]
Hello @Gcenx 

Thank you for your comments.  I believe I have tried what you suggested.  I wasn't sure how to installed faudio but in wintricks I did a search and under dll I found all the faudio ones.  so I chose the one that said "latest" - it said successful. 

This didn't work however and I still got the message before game start "no audio detected and video card not supported"

My engine is WS11WineCX64Bit19.01 so I went back into winetricks and choose faudio 19.01.  Did not work again with same behaviour.  Strange how I get the intro of the game though?

Here are the latest logs which I have tried to remove dups.  

<img width="517" alt="Screen Shot 2020-04-26 at 12 06 36 PM" src="https://user-images.githubusercontent.com/20375753/80313019-7847f200-87b6-11ea-9d6e-0f7d19c323bc.png">

000b:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.

001d:err:plugplay:process_IOService_Device object 0x4403

001d:err:plugplay:process_IOService_Device Unable to create plug in interface for USB deviceobject 0x7203
001d:err:plugplay:process_IOService_Device object 0x8103
001f:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0009:fixme:heap:RtlSetHeapInformation 0x0 1 0x0 0 stub
002f:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0033:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:heap:RtlSetHeapInformation 0x0 1 0x0 0 stub
0035:err:dbghelp_stabs:stabs_parse Unknown stab type 0x0a

0035:err:dbghelp_stabs:stabs_parse Unknown stab type 0x0a
0035:fixme:dbghelp_dwarf:dwarf2_parse_subprogram Unhandled Tag type 0x1 at ctx(0x32ebb0,L"d3dx9_43"), for debug_info(abbrev:0xb586f54,symt:0xb498470)
003f:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0040:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0041:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:wbemprox:enum_class_object_Next timeout not supported
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled

0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0BA6E3E8, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xbfd7000, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32de5c, modes 0x0 partial stub!
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xbfd7000, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32de5c, modes 0xf373990 partial stub!
0035:fixme:d3d:wined3d_check_device_multisample_type multisample_type 24 not handled yet.
0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0BEB2CF0, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled

0066:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0067:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0068:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled

0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0BFD65E8, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0x1116f0, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32db4c, modes 0x0 partial stub!
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0x1116f0, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32db4c, modes 0xf374248 partial stub!
0035:fixme:d3d:wined3d_check_device_multisample_type multisample_type 24 not handled yet.
0035:fixme:process:GetProcessWorkingSetSizeEx (0xffffffff,0x32dd50,0x32dd54,0x0): stub
006d:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
0035:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
0035:fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
0035:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
0035:fixme:driver:D3DKMTOpenAdapterFromHdc (0x32dc78): stub
0035:fixme:thread:SetThreadIdealProcessor (0x584): stub
0035:err:ole:COMPOBJ_DllList_Add couldn't find function DllGetClassObject in L"C:\\windows\\system32\\xaudio2_7.dll"
0035:err:ole:CoGetClassObject no class object {5a508685-a254-4fba-9b82-9a24b00306af} could be created for context 0x1
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:err:d3dcompiler:compile_shader HLSL shader parsing failed.

003b:fixme:hlsl_parser:hlsl_parse Check for valued return on void function.
003b:fixme:hlsl_parser:hlsl_parse Implicit conversion to the return type if needed, error out if conversion not possible.
003b:err:d3dcompiler:compile_shader HLSL shader parsing failed.
006e:fixme:d3d:state_linepattern_w Setting line patterns is not supported in OpenGL core contexts.
0074:fixme:winsock:WSALookupServiceBeginA (0x3410feec 0x00000ff0 0x3410ff34) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3411fee4 0x00000ff0 0x3411ff38) Stub!
007a:fixme:coreaudio:AudioSessionControl_RegisterAudioSessionNotification (0x2ba8b7f8)->(0x36552bb0) - stub
007a:err:ole:COMPOBJ_DllList_Add couldn't find function DllGetClassObject in L"C:\\windows\\system32\\xaudio2_7.dll"
007a:err:ole:CoGetClassObject no class object {5a508685-a254-4fba-9b82-9a24b00306af} could be created for context 0x1
007a:fixme:coreaudio:AudioSessionControl_UnregisterAudioSessionNotification (0x2ba8b7f8)->(0x36552bb0) - stub
0074:fixme:winsock:WSANSPIoctl (0x0, 0x88000019, 0x0, 0x00000000, 0x0, 0x00000000, 0x3411ff34, 0x3411ff20) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3410feec 0x00000ff0 0x3410ff34) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3411fee4 0x00000ff0 0x3411ff38) Stub!
0074:fixme:winsock:WSANSPIoctl (0x0, 0x88000019, 0x0, 0x00000000, 0x0, 0x00000000, 0x3411ff34, 0x3411ff20) Stub!

0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3410feec 0x00000ff0 0x3410ff34) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3411fee4 0x00000ff0 0x3411ff38) Stub!
0035:fixme:toolhelp:CreateToolhelp32Snapshot Unimplemented: heap list snapshot
0035:fixme:toolhelp:Heap32ListFirst : stub
0075:fixme:winsock:set_dont_fragment IP_DONTFRAGMENT for IPv4 not supported in this platform
0074:fixme:winsock:WSANSPIoctl (0x0, 0x88000019, 0x0, 0x00000000, 0x0, 0x00000000, 0x3411ff34, 0x3411ff20) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3410feec 0x00000ff0 0x3410ff34) Stub!
0074:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0074:fixme:winsock:WSALookupServiceBeginA (0x3411fee4 0x00000ff0 0x3411ff38) Stub!
wine: Unhandled exception 0xc0000409 in thread 75 at address 0x211d847 (thread 0075), starting debugger...
0075:err:seh:start_debugger Couldn't start debugger ("false") (2)
Read the Wine Developers Guide on how to set up winedbg or another debugger

**Comment by Gcenx** [comment]
Seems that game is attempting to launch ins DirectX11 mode, you need to force it into DirectX9 mode via the launcher option or editing its config file.

The other annoying one is it seems that version of FAudio is still missing some required functions the game requires, but not having that game in my library makes it difficult attempt to debug.

**Comment by gstlouisgit** [comment]
Thanks again for the reply @Gcenx 
sounds like this will not be an easy fix.  Unfortunately I don't know how to fore directx9 via launcher or its config file?

I did try with winetricks and found DirectX9 and ran that.  It didn't do anything more... 

The audio I wouldn't know where to begin with that either.

This is the latest log.  Where did you find out it was using direectx11 on there?   This way I can continue Tshooting without always annoying you with the same log possibly..

logs:

000b:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
000d:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0010:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0017:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
001d:err:plugplay:process_IOService_Device object 0x4503
001d:err:plugplay:process_IOService_Device object 0x5a03
001d:err:plugplay:process_IOService_Device Unable to create plug in interface for USB deviceobject 0x8803
001d:err:plugplay:process_IOService_Device object 0x8403
001f:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0009:fixme:heap:RtlSetHeapInformation 0x0 1 0x0 0 stub
002f:fixme:thread:create_user_shared_data_thread Creating user shared data update thread.
0033:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:heap:RtlSetHeapInformation 0x0 1 0x0 0 stub
0035:err:dbghelp_stabs:stabs_parse Unknown stab type 0x0a
0035:err:dbghelp_stabs:stabs_parse Unknown stab type 0x0a
003f:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0040:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:wbemprox:enum_class_object_Next timeout not supported
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled
0035:err:avicap:query_video_device Video 4 Linux support not enabled

0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0B7ED330, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xba724c8, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32de5c, modes 0x0 partial stub!
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xba724c8, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32de5c, modes 0xeed3990 partial stub!
0035:fixme:d3d:wined3d_check_device_multisample_type multisample_type 24 not handled yet.
2020-05-04 09:44:13.594 battlefield548338190Wine-preloader[22454:15402811] IMKInputSession [0x68458650 presentFunctionRowItemTextInputViewWithEndpoint:completionHandler:] : [self textInputContext]=0x682204d0 *NO* NSRemoteViewController to client, NSError=Error Domain=NSCocoaErrorDomain Code=4099 "The connection from pid 0 was invalidated from this process." UserInfo={NSDebugDescription=The connection from pid 0 was invalidated from this process.}, com.apple.inputmethod.EmojiFunctionRowItem
0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0B7EA760, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled
0035:err:avicap:query_video_device Video 4 Linux support not enabled

0066:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0067:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0068:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:wbemprox:wbem_services_CreateInstanceEnum unsupported flags 0x00000030
0035:fixme:ddraw:ddraw7_Initialize Ignoring guid {aeb2cdd4-6e41-43ea-941c-8361cc760781}.
0035:err:avicap:query_video_device Video 4 Linux support not enabled
0035:fixme:wbemprox:client_security_SetBlanket 6795D6B8, 0BA71AB0, 10, 0, (null), 3, 3, 00000000, 0x00000000
0035:fixme:wbemprox:client_security_Release 6795D6B8
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xb7ef588, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32db4c, modes 0x0 partial stub!
0035:fixme:dxgi:dxgi_output_GetDisplayModeList iface 0xb7ef588, format DXGI_FORMAT_R8G8B8A8_UNORM, flags 0x1, mode_count 0x32db4c, modes 0xeed4248 partial stub!
0035:fixme:d3d:wined3d_check_device_multisample_type multisample_type 24 not handled yet.
0035:fixme:process:GetProcessWorkingSetSizeEx (0xffffffff,0x32dd50,0x32dd54,0x0): stub
006d:fixme:d3d:wined3d_context_gl_init OpenGL implementation does not support GL_PRIMITIVE_RESTART_FIXED_INDEX.
0035:fixme:d3d:debug_d3dformat Unrecognized 0x34324644 (as fourcc: DF24) WINED3DFORMAT!
0035:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x34324644) in the format lookup table.
0035:fixme:d3d:debug_d3dformat Unrecognized 0x36314644 (as fourcc: DF16) WINED3DFORMAT!
0035:fixme:d3d:wined3d_get_format Can't find format unrecognized (0x36314644) in the format lookup table.
0035:fixme:driver:D3DKMTOpenAdapterFromHdc (0x32dc78): stub
0035:fixme:thread:SetThreadIdealProcessor (0x57c): stub
0035:err:ole:COMPOBJ_DllList_Add couldn't find function DllGetClassObject in L"C:\\windows\\system32\\xaudio2_7.dll"
0035:err:ole:CoGetClassObject no class object {5a508685-a254-4fba-9b82-9a24b00306af} could be created for context 0x1
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:err:d3dcompiler:compile_shader HLSL shader parsing failed.
003b:fixme:hlsl_parser:declare_vars Complex initializers are not supported yet.
003b:err:d3dcompiler:compile_shader HLSL shader parsing failed.
003b:fixme:hlsl_parser:hlsl_parse Check for valued return on void function.
003b:fixme:hlsl_parser:hlsl_parse Implicit conversion to the return type if needed, error out if conversion not possible.
003b:err:d3dcompiler:compile_shader HLSL shader parsing failed.
006e:fixme:d3d:state_linepattern_w Setting line patterns is not supported in OpenGL core contexts.
0074:fixme:thread:SetThreadIdealProcessor (0xfffffffe): stub
0075:fixme:winsock:WSALookupServiceBeginA (0x33f8feec 0x00000ff0 0x33f8ff34) Stub!
0075:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0075:fixme:winsock:WSALookupServiceBeginA (0x33f9fee4 0x00000ff0 0x33f9ff38) Stub!
007b:fixme:coreaudio:AudioSessionControl_RegisterAudioSessionNotification (0x2b499850)->(0x35998860) - stub
007b:err:ole:COMPOBJ_DllList_Add couldn't find function DllGetClassObject in L"C:\\windows\\system32\\xaudio2_7.dll"
007b:err:ole:CoGetClassObject no class object {5a508685-a254-4fba-9b82-9a24b00306af} could be created for context 0x1
007b:fixme:coreaudio:AudioSessionControl_UnregisterAudioSessionNotification (0x2b499850)->(0x35998860) - stub
0075:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0075:fixme:winsock:WSALookupServiceBeginA (0x33f9fee4 0x00000ff0 0x33f9ff38) Stub!
0075:fixme:winsock:WSANSPIoctl (0x0, 0x88000019, 0x0, 0x00000000, 0x0, 0x00000000, 0x33f9ff34, 0x33f9ff20) Stub!
0075:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0075:fixme:winsock:WSALookupServiceBeginA (0x33f8feec 0x00000ff0 0x33f8ff34) Stub!
0075:fixme:winsock:WSALookupServiceEnd (0x0) Stub!
0075:fixme:winsock:WSALookupServiceBeginA (0x33f9fee4 0x00000ff0 0x33f9ff38) Stub!
0035:fixme:toolhelp:CreateToolhelp32Snapshot Unimplemented: heap list snapshot
0035:fixme:toolhelp:Heap32ListFirst : stub
0076:fixme:winsock:set_dont_fragment IP_DONTFRAGMENT for IPv4 not supported in this platform
wine: Unhandled exception 0xc0000409 in thread 76 at address 0x211d847 (thread 0076), starting debugger...
0076:err:seh:start_debugger Couldn't start debugger ("false") (2)
Read the Wine Developers Guide on how to set up winedbg or another debugger

**Comment by Gcenx** [comment]
The `dxgi` points to DirectX10/11 usage I then googled for the DirectX version used.

This information from [Steam forms](https://steamcommunity.com/app/235600/discussions/0/846962626940571418/) explains editing a config file to force DirectX9 mode.

Still the other issue is still FAudio seems the old PE version on winetricks is missing something that’s required for this to run still, maybe swapping to using xact would be better in this case

**Comment by Jrogan23** [comment]
Hi, I have Catalina 10.15.4. I have downloaded the most recently updated wineskin winery files, and I tried to install the most recent WS11WineCX64Bit19.01 engine. It takes a bit to process like it is downloading, but then it just takes me back to the first window and there is nothing under installed engines and I have no idea where it has gone or what has happened. I downloaded the wineskin winery files first, then I entered into recovery mode on my mac and entered nvram boot-args="no32exec=0" into the terminal. Should I have done this before downloading the files? I thought it was only necessary to do before downloading the engine. I am not very familiar with terminal, but when I opened it, it said "-bash-3.2#" which it kept repeating when I pressed entered after entering nvram boot-args="no32exec=0".

Anybody have any clues what might be happening? The only thing that stood out to me is the box that says "hide engines that require Xquartz" is checked, and I can't uncheck it. But I understand this is part of the new update? Maybe there is an accompanying file that I haven't extracted properly?
